tests/unit/engine/rest/test_rest_engine.py

```diff
@@ -109,3 +120,123 @@ def test_request_params():
         for sq_params in combinations(testing_params, n_sq_params):
             req_params = {p: 'override' for p in sq_params}
             validate_args(engine, req_params)
+
+
+
+@pytest.mark.rest
+@pytest.mark.skipif(not os.environ.get('TEST_SERVER', None),
+                    reason='causes github action hang')
+def test_server_cert():
+    '''Can we can get a valid response with & without certificate'''
+    from tempfile import mkstemp
+
+    # We need to change the port used to avoid conflicts
+    config = {'data-directory': './tests/data/parquet',
+              'temp-directory': '/tmp/suzieq',
+              'logging-level': 'WARNING',
+              'test_set': 'basic_dual_bgp',  # an extra field for testing
+              'rest': {
+                'rest-certfile': './tests/test_cert_CA/server-cert.pem',
+                'rest-keyfile': './tests/test_cert_CA/server-key.pem',
+                'API_KEY': '496157e6e869ef7f3d6ecb24a6f6d847b224ee4f',
+                'logging-level': 'WARNING',
+                'address': '0.0.0.0',
+                'port': get_free_port(),
+                'no-https': True,
+                'log-stdout': True
+              },
+              'analyzer': {'timezone': 'GMT'},
+              }
+
+    def create_config(config):
+        fd, tmpfname = mkstemp(suffix='.yml')
+        f = os.fdopen(fd, 'w')
+        f.write(yaml.dump(config))
+        f.close()
+
+        cfgfile = tmpfname
+        sqcfg = load_sq_config(config_file=cfgfile)
+
+        print(f'sqcfg: {sqcfg}')
+
+        with open(cfgfile, 'w') as f:
+            f.write(yaml.safe_dump(sqcfg))
+        return sqcfg, cfgfile
+
+    def open_rest_server(cfgfile):
+        server_cmd_args = f'{suzieq_rest_server_path} -c {cfgfile}'.split()
+        # pylint: disable=consider-using-with
+        proc = subprocess.Popen(server_cmd_args)
+        sleep(5)
+        return proc
+
+    def make_get_response_request(sqcfg):
+        ctxt = SqContextMock(
+        rest_api_key=sqcfg['rest']['API_KEY'],
+        rest_transport= \
+            'http' if sqcfg['rest']['no-https'] == True else 'https',
+        rest_server_ip=sqcfg['rest']['address'],
+        rest_server_port=sqcfg['rest']['port'],
+        cfg={'rest': sqcfg['rest']})
+
+        sqobj = SqObjMock(ctxt, '', '', 'default',
+                        'default', 'latest', 'device', 'default')
+
+        print(ctxt)                                                                           
+        print(sqobj)
+        engine = SqRestEngine(sqobj)
+        try:
+            response = engine._get_response('show')
+            print(f'responsein test: {response}')
+            return 200
+        except:
```

Comment:
Which kind of exception are you trying to catch? If that isn't clear use:
```suggestion
        except Exception:
```